lib/order.rb

```diff
@@ -0,0 +1,84 @@
+require 'twilio-ruby'
+require_relative '../.env.rb'
+
+class Order 
+
+  attr_accessor :basket
+  attr_reader :total, :menu
+
+  def initialize(menu = Menu.new)
+    @menu = menu
+    @basket = {}
+    @total = 0.0
+  end
+
+  def check_dishes
+    list_of_dishes = ''
+    menu.dishes_available.each do |dish, price|
+      list_of_dishes += ("#{dish} £#{price}\n")
+    end
+    list_of_dishes
+  end
+
+  def add_item(dish, quantity = 1)
+    fail 'That item is not on the menu' unless menu.dishes_available.has_key?(dish)
+    dish_check_and_add(dish, quantity)
+    price = quantity * menu.dishes_available[dish]
+    @total += price
+  end
+
+  def remove_item(dish, quantity = 1)
+    fail 'Item was not in the basket' unless basket.include?(dish)
+    fail 'You do not have that quantity of the item in the basket' if quantity > basket[dish]
+    quantity_check_and_remove(dish, quantity)
+    price = quantity * menu.dishes_available[dish]
+    @total -= price
+  end
+
+  def total_price_verified?
+    basket_total = 0.0
+    basket.each do |dish, quantity|
+      basket_total += quantity * menu.dishes_available[dish]
+    end
+    basket_total == total ? true : false
+  end
+
+  def place_order
+    fail 'The total price has been miscalculated' unless total_price_verified?
+    send_message
+    "Thank you for your order. You will receive text confirmation shortly"
+  end
+
+  private 
+
+  def dish_check_and_add(dish, quantity)
+    basket.has_key?(dish) ? basket[dish] += quantity : basket[dish] = quantity
```

Comment:
Style/DeprecatedHashMethods: `Hash#has_key?` is deprecated in favor of `Hash#key?`.